cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -1267,6 +1267,18 @@ impl MachInstEmit for Inst {
 
                 put_string(&format!("$ => {} :{opcode}\n", reg_name(rd.to_reg())), sink);
 
+                if cc == IntCC::NotEqual {
+                    if rd.to_reg() == a0() {
+                        put_string("1 => B\n", sink);
+                    } else if rd.to_reg() == b0() {
+                        put_string("1 => A\n", sink);
+                    } else {
+                        put_string(&format!("{} => A\n", reg_name(rd.to_reg())), sink);
```

Comment:
Make the regalloc allocate `A` or `B` as the output register instead.

Reply:
Aaaand, is it always a good way? As I see during compilation of some progrmas, firstly regalloc tries use ``A`` as output, than, if it is diffuclt, ``B``, and only than ``C``. So, we optimizing 1 register-register copying, but enforce regalloc to use ``A`` or ``B`` in situation when it very don't want to do this. I'm not saying I'm right, but for me is not obvious that this approach is better

Reply:
It is always better to provide the exact information to regalloc – if it can’t satisfy the requirements it has an ability to decide the best location to introduce a register spill, and which register to spill. Clobbering the register for this specific instruction removes that option.

Reply:
Admittedly the fact that many instructions work on A and B registers makes this somewhat harder to justify, but regalloc still has much better overview of the function it is genereating to make an ideal decision. Even if it does not do so right this moment for the specific code samples you have on hand.

Reply:
Hmmm, thank you

Reply:
Fixed [here](https://github.com/near/wasmtime/pull/57/commits/20e4501671f03e54410cb9b2df81acdad2025761)